### PR TITLE
Feature/services login auth

### PR DESCRIPTION
**Pull request by jesusdavid24**
**Refactoring schema prism schema and testing of login endpoints.

Updated relationship between `facilities`, `order`, and `restaurant` tables and User Endpoint Testing

### Description of Changes

In this Pull Request, I have made the following changes:

1. **Migration in Prism:** A new migration has been created in Prism to update the relationship between the `facilities` table and the `order` and `restaurant` tables. This update was necessary to ensure that each order is linked to a specific restaurant facility. This is crucial for tracking purchases by location.

2. **User Endpoint Testing:** Endpoints related to user creation, deletion, update and query have been extensively tested. Testing is essential to ensure the proper functioning of these functionalities.

### Testing Details

The following user endpoints have been extensively tested:

- User creation.
- User deletion.
- User information update.
- User queries.

### CODE REVIEW

**Review by ht1204 — APPROVED** [review]
Great Job.

**Review by Cristianjs93 — CHANGES_REQUESTED** [review]
image

Please remove commented code from file users/index.ts

**Review by ht1204 — APPROVED** [review]

**Review by Cristianjs93 — APPROVED** [review]

File: src/api/users/user.controller.ts [hunk 1]
```diff
@@ -1,110 +1,104 @@
...
-      res.status(200).json(users);
-
-   } catch (error: any) {
-      res.status(400).send({ message: error.message });
-   }
+  try {
+    const users = await getAllUsers();
+    res.status(200).send(users);
+  } catch (ex: any) {
+    res.status(400).send(ex.toSting());
```
Comment (ht1204) [inline src/api/users/user.controller.ts:19]: Checking this typo, which should be `toString()`, why not consider this error management?
 ```ts
catch (e: unknown) { // <-- note `e` has explicit `unknown` type
    e.message // errors
    if (typeof e === "string") {
        e.toUpperCase() // works, `e` narrowed to string
    } else if (e instanceof Error) {
        e.message // works, `e` narrowed to Error
    }
    // ... handle other error types 
}
```

File: src/api/users/user.controller.ts [hunk 2]
```diff
@@ -1,110 +1,104 @@
...
+
+    if (!user) {
+      return res.status(404).json({
+        message: 'User not found',
+      });
+    }
+
+    return res.status(200).send(user);
+  } catch (ex: any) {
+    return res.status(400).send(ex.toSting());
```
Comment (ht1204) [inline src/api/users/user.controller.ts:37]: same here.

File: src/api/users/user.controller.ts [hunk 3]
```diff
@@ -1,110 +1,104 @@
...
+    const user = await createUser(data);
 
-      res.status(201).json(user);
-
-   } catch (error: any) {
-      res.status(400).send({ message: error.message });
-   }
+    return res.status(201).json(user);
+  } catch (ex: any) {
+    return res.status(400).send(ex.toSting());
```
Comment (ht1204) [inline src/api/users/user.controller.ts:67]: and here.

File: src/api/users/user.controller.ts [hunk 4]
```diff
@@ -1,110 +1,104 @@
...
+        message: 'User not found',
+      });
+    }
 
-   } catch (error: any) {
-      res.status(400).send({ message: error.message });
-   }
+    return res.status(200).json(user);
+  } catch (ex: any) {
+    return res.status(400).send(ex.toSting());
```
Comment (ht1204) [inline src/api/users/user.controller.ts:85]: and here.

File: src/api/users/user.services.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import { PrismaClient } from '@prisma/client';
 
-import { hashPassword } from '../../auth/utils/bycript';
-import { Users } from './user.types';
+import { hashPassword } from '../../auth/utils/bycript.ts';
+import { Users } from './user.types.ts';
```
Comment (ht1204) [inline src/api/users/user.services.ts:4]: is there any issue in importing files? ts file extension should not be necessary for adding.

**Pull request merged**